src/chainparams.cpp

```diff
@@ -619,3 +620,32 @@ void TurnOffBIP65() {
 void TurnOffBIP66() {
 	globalChainParams->TurnOffBIP66();
 }
+
+
+Consensus::NetworkPeriod GetNetworkPeriodForBlock(const Consensus::Params& params, int nBlockHeight)
+{
+    int period = 0;
+    for (; period < Consensus::MAX_NETWORK_PERIOD; ++period)

```

Comment:
it would be better in my opinion

`for (Consensus::NetworkPeriod p = Consensus::MAX_NETWORK_PERIOD; p > NETWORK_PERIOD_X16R; --p)
{
    if (nBlockHeight > p)
    {
         return p;
    }
}

return NETWORK_PERIOD_X16R;`

Reply:
+